PWGCF/TwoParticleCorrelations/Tasks/r2p2-4-id.cxx

```diff
@@ -0,0 +1,444 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/DataModel/PIDResponse.h"
+#include "Framework/ASoAHelpers.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+//-----PID Functions-----------------------------------------------------------
+template <typename T>
+bool NO_PID(T track)
+{
+  return true;
+}
+template <typename T>
+bool PID_PION(T track)
+{
+  float tpccut = 2.5, tofcut = 2.5;
+  if (fabs(track.tpcNSigmaPi()) >= tpccut)
+    return false;
+  if (track.pt() >= 0.6) {
+    if (track.hasTOF()) {
+      if (fabs(track.tofNSigmaPi()) >= tofcut)
+        return false;
+    } else {
+      return false;
+    }
+  }
+  return true;
+}
+template <typename T>
+bool PID_KAON(T track)
+{
+  float tpccut = 2, tofcut = 2;
+  if (track.pt() < 0.6) {
+    if (track.pt() < 0.45)
+      tpccut = 2;
+    else if (track.pt() < 0.55)
+      tpccut = 1;
+    else
+      tpccut = 0.6;
+    if (fabs(track.tpcNSigmaKa()) > tpccut)
+      return false;
+  } else if (track.hasTOF()) {
+    if ((fabs(track.tpcNSigmaKa()) > tpccut) || (fabs(track.tofNSigmaKa()) > tofcut))
+      return false;
+  } else {
+    return false;
+  }
+  return true;
+}
+template <typename T>
+bool PID_PROTON(T track)
+{
+  float tpccut = 2.2, tofcut = 2;
+  if (track.pt() < 1.1) {
+    if (track.pt() < 0.85)
+      tpccut = 2.2;
+    else
+      tpccut = 1;
+    if (fabs(track.tpcNSigmaPr()) > tpccut)
+      return false;
+  } else if (track.hasTOF()) {
+    if ((fabs(track.tpcNSigmaPr()) > tpccut) || (fabs(track.tofNSigmaPr()) > tofcut))
+      return false;
+  } else {
+    return false;
+  }
+  return true;
+}
+//-----------------------------------------------------------------------------
+template <typename T>
+bool (*pidarray[4])(T) = {NO_PID, PID_PION, PID_KAON, PID_PROTON}; // Array of PID functions
+
+namespace o2::aod
+{
+namespace idr2p2columns
+{
+DECLARE_SOA_COLUMN(BinFlag, binflag, bool); // To flag tracks without proper binning
+DECLARE_SOA_COLUMN(PID, pid, int);          // Flag for type of particle
```

Comment:
Please, one single column. Use a negative value when not accepted
Also probably int is too large. See my below comments

Reply:
Ok, I'll change the datatype to int8_t